SecretSanta/src/SecretSanta.Api/Controllers/GiftController.cs

```diff
@@ -2,33 +2,74 @@
 using System.Collections.Generic;
 using System.Linq;
 using Microsoft.AspNetCore.Mvc;
-using SecretSanta.Domain.Models;
+using SecretSanta.Api.DTO;
 using SecretSanta.Domain.Services;
 
 namespace SecretSanta.Api.Controllers
 {
-    [Route("api/[controller]")]
-    [ApiController]
-    public class GiftController : ControllerBase
-    {
-        private readonly IGiftService _GiftService;
-
-        public GiftController(IGiftService giftService)
-        {
-            _GiftService = giftService ?? throw new ArgumentNullException(nameof(giftService));
-        }
-
-        // GET api/Gift/5
-        [HttpGet("{userId}")]
-        public ActionResult<List<DTO.Gift>> GetGiftForUser(int userId)
-        {
-            if (userId <= 0)
-            {
-                return NotFound();
-            }
-            List<Gift> databaseUsers = _GiftService.GetGiftsForUser(userId);
-
-            return databaseUsers.Select(x => new DTO.Gift(x)).ToList();
-        }
-    }
-}
+	[Route("api/[controller]")]
+	[ApiController]
+	public class GiftController : ControllerBase
+	{
+		private readonly IGiftService _giftService;
+
+		public GiftController(IGiftService giftService)
+		{
+			_giftService = giftService ?? throw new ArgumentNullException(nameof(giftService));
+		}
+
+		// GET api/Gift/5
+		[HttpGet("{userId}")]
+		public ActionResult<List<Gift>> GetGiftsForUser(int userId)
+		{
+			if (userId <= 0) return BadRequest("userId invalid.");
+
+			return _giftService.GetGiftsForUser(userId)
+				.Select(modelGift => new Gift(modelGift))
+				.ToList();
+		}
+
+		// POST api/Gift/5
+		[HttpPost("{userId}")]
+		public ActionResult<List<Gift>> AddGiftsToUser(int userId, List<Gift> gifts)
+		{
+			if (gifts is null) return BadRequest("gifts is null.");
+			if (userId <= 0) return BadRequest("userId invalid.");
```

Comment:
The messages here could benefit from using `nameof`
```suggestion
			if (userId <= 0) return BadRequest($"{nameof(userId)} invalid.");
```